src/filtrering/filtrering-mine-filter/dragAndDrop/keyboard-event-listeners.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {keyCodes} from '../../../utils/utils';
+
+export interface HandleKeyDownProps {
+    eventIsInsideContainer: (e: React.MouseEvent) => boolean;
+    setDropIndex: React.Dispatch<React.SetStateAction<number>>;
+    prepFlyttOpp: (index: number) => void;
+    prepFlyttNed: (index: number) => void;
+    requestFocus: (index: number) => void;
+}
+export interface HandleKeyUpProps {
+    eventIsInsideContainer: (e: React.MouseEvent) => boolean;
+    requestNewOrder: (from: number, to: number) => void;
+    lagreRekkefølge: () => void;
+    avbryt: () => void;
+    setAriaTekst: React.Dispatch<React.SetStateAction<string>>;
+    setDestIndex: React.Dispatch<React.SetStateAction<number>>;
+    setSrcIndex: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export function handleKeyDown({
+    eventIsInsideContainer,
+    setDropIndex,
+    prepFlyttOpp,
+    prepFlyttNed,
+    requestFocus
+}: HandleKeyDownProps) {
+    return (e) => {
+        if (eventIsInsideContainer(e)) {
+            if (e.keyCode === keyCodes.down || e.keyCode === keyCodes.up) {
+                e.preventDefault();
+                if (e.keyCode === keyCodes.up) {
+                    if (e.altKey) {
+                        prepFlyttOpp(e.target.value);
+                    } else {
+                        requestFocus(e.target.value - 1);
+                    }
+                } else {
+                    if (e.altKey) {
+                        prepFlyttNed(e.target.value);
+                    } else {
+                        requestFocus(e.target.value + 1);
+                    }
+                }
+                setDropIndex(-1);
+            } else if (e.keyCode === keyCodes.space) e.preventDefault();
+        }
+    };
+}
+
+export function handleKeyUp({
+    eventIsInsideContainer,
+    requestNewOrder,
+    setAriaTekst,
+    setDestIndex,
+    setSrcIndex,
+    avbryt,
+    lagreRekkefølge
+}: HandleKeyUpProps) {
+    return (e) => {
+        if (eventIsInsideContainer(e)) {
+            if (e.altKey && (e.keyCode === keyCodes.up || e.keyCode === keyCodes.down)) {
+                if (e.keyCode === keyCodes.up) {
+                    requestNewOrder(e.target.value, e.target.value - 1);
+                } else {
+                    requestNewOrder(e.target.value, e.target.value + 1);
+                }
+            } else if (e.keyCode === keyCodes.space) {
+                setAriaTekst(
+                    'Bruk piltast opp eller ned for å velge et annet filter. Hold nede alt og press opp eller ned for å endre rekkefølgen til filter. Enter for å lagre. Escape for å avbryte.'
```

Comment:
Småpirk, men "alt-knappen" ? Siden "alt" betyr noe annet enn bare en ting på norsk? :P 

'Bruk piltast opp eller ned for å velge et annet filter. Hold nede alt-knappen og trykk opp eller ned for å endre rekkefølgen til filteret. Trykk enter for å lagre, og escape for å avbryte.' For eksempel? Kari er en bedre tekstforfatter, men ja. Må være tydelig

Reply:
Vel det var dette som ikke kom opp for David. Kan jo egentlig fjerne det?

Reply:
Hmm men David brukte noe annet enn vanlig reader? Tenker at vi ikke skal fjerne det, siden det er kanskje noen som bruker "vanlig" reader!